packages/raf/src/index.ts

```diff
@@ -36,7 +38,116 @@ function createRAF(
   };
   const stop = () => {
     setRunning(false);
-    cancelAnimationFrame(requestID);
+    if (requestID !== null) cancelAnimationFrame(requestID);
+  };
+
+  onCleanup(stop);
+  return [running, start, stop];
+}
+
+/**
+ * Returns an advanced primitive factory function (that has an API similar to `createRAF`) to handle multiple animation frame callbacks in a single batched `requestAnimationFrame`, avoiding the overhead of scheduling multiple animation frames outside of a batch and making them all sync on the same delta.
+ *
+ * This is a [singleton root](https://github.com/solidjs-community/solid-primitives/tree/main/packages/rootless#createSingletonRoot) primitive.
+ *
+ * @returns Returns a factory function that works like `createRAF` but handles all scheduling in the same frame batch and optionally automatically starts and stops the global loop.
+ * ```ts
+ * (callback: FrameRequestCallback, automatic?: boolean) => [queued: Accessor<boolean>, queue: VoidFunction, dequeue: VoidFunction, running: Accessor<boolean>, start: VoidFunction, stop: VoidFunction]
+ * ```
+ *
+ * @example
+ * const createScheduledFrame = useFrameloop();
+ *
+ * const [queued, queue, dequeue, running, start, stop] = createScheduledFrame(() => {
+ *   el.style.transform = "translateX(...)"
+ * });
+ */
+const useFrameloop = createHydratableSingletonRoot<
+  (
+    callback: FrameRequestCallback,
+    automatic?: MaybeAccessor<boolean>,
+  ) => [
+    queued: Accessor<boolean>,
+    queue: VoidFunction,
+    dequeue: VoidFunction,
+    running: Accessor<boolean>,
+    start: VoidFunction,
+    stop: VoidFunction,
+  ]
+>(() => {
+  if (isServer) return () => [() => false, noop, noop, () => false, noop, noop];
+
+  const frameCallbacks = new ReactiveSet<FrameRequestCallback>();
+
+  const [running, start, stop] = createRAF(delta => {
+    frameCallbacks.forEach(frameCallback => {
+      frameCallback(delta);
+    });
+  });
+
+  return function createFrame(callback: FrameRequestCallback, automatic = false) {
+    const queued = () => frameCallbacks.has(callback);
+    const queue = () => {
+      frameCallbacks.add(callback);
+      if (access(automatic) && !running()) start();
+    };
+    const dequeue = () => {
+      frameCallbacks.delete(callback);
+      if (running() && frameCallbacks.size === 0) stop();
+    };
+
+    onCleanup(dequeue);
+    return [queued, queue, dequeue, running, start, stop];
+  };
+});
+
+/**
+ * An advanced primitive creating reactive scheduled frameloops, for example [motion's frame util](https://motion.dev/docs/frame), that are automatically disposed onCleanup.
+ *
+ * The idea behind this is for more complex use cases, where you need scheduling and want to avoid potential issues arising from running more than one `requestAnimationFrame`.
+ *
+ * @see https://github.com/solidjs-community/solid-primitives/tree/main/packages/raf#createScheduledFrameloop
```

Comment:
You seem to have forgotten to update the README.md

Reply:
Yes, sorry for that!